File: src/main/java/ch/salon/domain/TimelineData.java
package ch.salon.domain;

import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class TimelineData {
    private String eventId;
    private int intervalMinutes;
    private List<TimelineDay> days = new ArrayList<>();
    private List<TimelineRoom> rooms = new ArrayList<>();
}

File: src/main/webapp/app/shared/components/talks-planning/model/timeline-data.ts
import {TimelineDay} from "./timeline-day";
import {TimelineRoom} from "./timeline-room";

import {IntervalMinutes} from "./interval-minutes";

export interface TimelineData {
    eventId: string;
    intervalMinutes: IntervalMinutes;
    days: TimelineDay[];
    rooms: TimelineRoom[];
}